packages/data-context/src/data/ProjectLifecycleManager.ts

```diff
@@ -844,13 +844,25 @@ export class ProjectLifecycleManager {
   addWatcher (file: string | string[]) {
     const w = chokidar.watch(file, {
       ignoreInitial: true,
+      cwd: this.ctx.currentProject || '.',
```

Comment:
File watcher wasn't detecting new files on CI unless cwd was specified.

Reply:
How can `this.ctx.currentProject` ever be null, though?

Reply:
It can't at this stage, this was just to appease the types. Looks like this file uses `this._projectRoot` rather than `this.ctx.currentProject` so I'll use that instead. It can still be null, but I'll add the `assert` similar to how it's done in other places.